PixelDriver.cpp

```diff
@@ -219,39 +265,46 @@ const uint8_t* ICACHE_RAM_ATTR PixelDriver::fillWS2811(const uint8_t *buff,
         enqueue(LOOKUP_2811[(GAMMA_TABLE[subpix] >> 2) & 0x3]);
         enqueue(LOOKUP_2811[GAMMA_TABLE[subpix] & 0x3]);
 
-        buff += 3;
-    }
+        if (color_count == 4) {
+            subpix = buff[wOffset];
+            enqueue(LOOKUP_2811[(GAMMA_TABLE[subpix] >> 6) & 0x3]);
+            enqueue(LOOKUP_2811[(GAMMA_TABLE[subpix] >> 4) & 0x3]);
+            enqueue(LOOKUP_2811[(GAMMA_TABLE[subpix] >> 2) & 0x3]);
+            enqueue(LOOKUP_2811[GAMMA_TABLE[subpix] & 0x3]);
+        }
 
+        buff += color_count;
+    }
     return buff;
 }
 
 void ICACHE_RAM_ATTR PixelDriver::show() {
     if (!pixdata) return;
 
-    if (type == PixelType::WS2811) {
+    if (type == PixelType::WS2811 || type == PixelType::SK6812RGBW) {
         if (!cntZigzag) {  // Normal / group copy
-            for (size_t led = 0; led < szBuffer / 3; led++) {
+            for (size_t led = 0; led < szBuffer / this->color_count; led++) {
                 uint16 modifier = led / cntGroup;
-                asyncdata[3 * led + 0] = pixdata[3 * modifier + 0];
-                asyncdata[3 * led + 1] = pixdata[3 * modifier + 1];
-                asyncdata[3 * led + 2] = pixdata[3 * modifier + 2];
+                for (uint8_t c = 0; c < this->color_count; c++) {
+                    asyncdata[this->color_count * led + c] = pixdata[this->color_count * modifier + c];
+                }
             }
         } else {  // Zigzag copy
```

Comment:
At this point, it may be beneficial to (at least via comment) indicate the else case is defined as `== PixelType::GECE`.